tfdml/core/dml_command_allocator_ring.h

```diff
@@ -30,41 +30,44 @@ class DmlCommandAllocatorRing
   public:
     DmlCommandAllocatorRing(
         ID3D12Device* device,
-        D3D12_COMMAND_LIST_TYPE command_list_type)
+        D3D12_COMMAND_LIST_TYPE command_list_type,
+        DmlGpuEvent initial_event)
     {
         for (auto& info : command_allocators_)
         {
             DML_CHECK_SUCCEEDED(device->CreateCommandAllocator(
                 command_list_type,
                 IID_PPV_ARGS(&info.allocator)));
+
+            info.completion_event = initial_event;
         }
     }
 
-    ID3D12CommandAllocator* GetCurrentAllocator()
+    ID3D12CommandAllocator* GetNextAllocator(DmlGpuEvent next_completion_event)
```

Comment:
[minor] Is the constructor heavyweight and owning anything? Is it better to pass by value and/or to `std::move` it on line 42?

(oi, that 80 column limit from the 1990's days of 640x480 monitors sure hinders readability on lines like line 48 - consider upgrading to 2022 someday)

Reply:
Very lightweight struct with only a default constructor.